openmc/mgxs_library.py

```diff
@@ -379,200 +436,587 @@ def tabular_legendre(self, tabular_legendre):
-            shape = (self._energy_groups.num_groups,)
-        elif self._representation is 'angle':
-            shape = (self._num_polar, self._num_azimuthal,
-                     self._energy_groups.num_groups)
+        """This method sets the total cross section by performing a
+        deep-copy of the provided ndarray. If the angular
+        representation is "isotropic" the shape of the input array
+        must be the number of energy groups. If the angular
+        representation is "angle" then the shape of the input
+        array must be the number of polar angles, number azimuthal
+        angles and energy groups.
+
+        Parameters
+        ----------
+        total: ndarray
+            Array of group-wise cross sections to apply
+
+        """
+
         # check we have a numpy list
-        check_type("total", total, np.ndarray, expected_iter_type=Real)
-        if total.shape == shape:
-            self._total = np.copy(total)
-        else:
-            msg = 'Shape of provided total "{0}" does not match shape ' \
-                  'required, "{1}"'.format(total.shape, shape)
-            raise ValueError(msg)
+        check_type('total', total, np.ndarray, expected_iter_type=Real)
+        # Check the dimensions of the data
+        check_value('total shape', total.shape, self.vector_shape)
+
+        self._total = np.copy(total)
 
     @absorption.setter
     def absorption(self, absorption):
-        if self._representation is 'isotropic':
-            shape = (self._energy_groups.num_groups,)
-        elif self._representation is 'angle':
-            shape = (self._num_polar, self._num_azimuthal,
-                     self._energy_groups.num_groups)
+        """This method sets the absorption cross section by performing a
+        deep-copy of the provided ndarray. If the angular
+        representation is "isotropic" the shape of the input array
+        must be the number of energy groups. If the angular
+        representation is "angle" then the shape of the input
+        array must be the number of polar angles, number azimuthal
+        angles and energy groups.
+
+        Parameters
+        ----------
+        absorption: ndarray
+            Array of group-wise cross sections to apply
+
+        """
+
         # check we have a numpy list
-        check_type("absorption", absorption, np.ndarray, expected_iter_type=Real)
-        if absorption.shape == shape:
-            self._absorption = np.copy(absorption)
-        else:
-            msg = 'Shape of provided absorption "{0}" does not match shape ' \
-                  'required, "{1}"'.format(absorption.shape, shape)
-            raise ValueError(msg)
+        check_type('absorption', absorption, np.ndarray,
+                   expected_iter_type=Real)
+        # Check the dimensions of the data
+        check_value('absorption shape', absorption.shape, self.vector_shape)
+
+        self._absorption = np.copy(absorption)
```

Comment:
Alright I give in, I've changed them.  I didnt have a strong story for why they shouldn be there, so I guess I'll save memory.